openlibrary/catalog/get_ia.py

```diff
@@ -154,8 +155,7 @@ def get_from_archive_bulk(locator):
 
     assert 0 < length < MAX_MARC_LENGTH
 
-    ureq = urllib.request.Request(url, None, {'Range': 'bytes=%d-%d' % (r0, r1)})
-    f = urlopen_keep_trying(ureq)
+    f = urlopen_keep_trying(url, headers={'Range': 'bytes=%d-%d' % (r0, r1)})
     data = None
     if f:
         data = f.read(MAX_MARC_LENGTH)
```

Comment:
```suggestion
        data = f[:MAX_MARC_LENGTH]
```
As discussed [here](https://github.com/internetarchive/openlibrary/pull/4388/files#r554017332) the previous code returned [`http.client.HTTPResponse`](https://docs.python.org/3/library/http.client.html#http.client.HTTPResponse) which has a `.read()` method that returns bytes.  The proposed code is trying to call `str.read()` which does not exist.

Reply:
The `response` is not indexable, `f` is a `requests.Response`.  Instead it might be better to use the `Response.raw` which is a `HTTPResponse` to use the exist code.  What do you think of this approach?

Reply:
@hornc can guide us here but my sense is that marc records should be bytes and we should do
`data = f.content[:MAX_MARC_LENGTH]`